### ISSUE TITLE
Create JitBuilder interface for defining types from existing C/C++ types

### ISSUE THREAD

**Issue by Leonardo2718** [eclipse/omr#445]
To simplify type definition/creation, it would be very useful to have a way of producing a JitBuilder `IlType` that matches a type specified by a user. For example:

```c++
TR::TypeDictionary td;
auto type = td.toIlType<int>::type; // could also potentially be `td.toIlType<int>()`
```

This would be particularly helpful in dealing with type aliases.

```c++
#ifdef USE_BIG_FLOAT
using Float = double;
#else
using Float = float;
#endif

TR::TypeDictionary td;
auto type = td.toIlType<Float>::type;
```

**Comment by mstoodle** [comment]
Note that Float != Double and float != double on 64-bit systems, so I'm not sure your example is something we necessarily want (unless the runtime you're working with wants 32-bit floats on 32-bit and 64-bit floats on 64-bit.

**Comment by Leonardo2718** [comment]
Yeah, all I wanted to show with that example is that in some cases, a runtime can be configured at compile time to use different types for some things. I'll change the condition name to something else to make it less confusing.

**Comment by Leonardo2718** [comment]
Solved by #528.
